runtime/syntax/gleam.yaml

```diff
@@ -0,0 +1,36 @@
+filetype: gleam
+
+detect:
+    filename: "\\.gleam$"
+
+rules:
+    - identifier: "\\b[a-z][a-z0-9_]*\\b"
+    - ignore: "(\\b(_[a-z][a-z0-9_]*)\\b|\\b(auto|delegate|derive|echo|else|implement|macro|test)\\b)"
+    - identifier.class: "[a-z][a-z0-9_]*\\("
+    - statement: "\\b(import|as|assert|case|const|if|let|panic|todo|use|opaque|pub|type|fn)\\b"
+    - type: "\\b[A-Z]+[A-Za-z0-9_]*\\b"
+    - constant.string:
+        start: '"'
+        end: '"'
+        skip: '\\"'
+        rules: []
+    # bit arrays
+    - special:
+        start: "<<"
+        end: ">>"
+        rules: []
+    - symbol: "(\\.\\.|\\.|,|:)"
+    - symbol.brackets: "({|}|\\(|\\)|\\[|\\]|<<|>>)"
+    - symbol.operator: "(^\\s+\\|>|<>|<-|->|\\+\\.|\\-\\.|/\\.|\\*\\.|%\\.|\\+|\\-|/|\\*|%|=)"
+    - constant.number: "\\b(0[bBoOxX][a-fA-F0-9][a-fA-F0-9_\\.]*|[0-9][0-9_\\.]*)\\b"
+    - constant.bool: "\\s(<|>|<\\.|>\\.|<=|>=|<=\\.|>=\\.|==|!=|==\\.|&&|\\|\\||\\|)\\s"
```

Comment:
Agreed, ideally there would be an `operator.bool` option to support the highlighting logic below, which I find helpful. I understand if it's too idiosyncratic for a default, though. Will fix the whitespace issue.

![image](https://github.com/zyedidia/micro/assets/1424402/37dab079-75ae-4118-8914-22fb3d0edd9a)
![image](https://github.com/zyedidia/micro/assets/1424402/a019436e-a487-45ad-a639-5dec09e67995)

Reply:
I wasn't able to remove the whitespace without clashing with the other operators... I'm afraid it's a bit brittle, but it does well with real code, at least using my color scheme.

Reply:
> Agreed, ideally there would be an `symbol.operator.bool` option to support the highlighting logic below, which I find helpful. I understand if it's too idiosyncratic for a default, though.

The way the highlighting works if you have `symbol.operator.bool` it will first look for `symbol.operator.bool`, then `symbol.operator`, and finally `symbol` in the current colorscheme. So that should just work, although you would need to use a customized colorscheme to see the difference.

> I wasn't able to remove the whitespace without clashing with the other operators... I'm afraid it's a bit brittle, but it does well with real code, at least using my color scheme.

You can usually work around the clashes by reordering the definitions (if I recall correctly the later ones take precedence). Although it's not much of an issue in practice because almost all colorschemes use the same color for the two different kinds of operators anyway.